lib/connection.js

```diff
@@ -99,10 +104,28 @@ function Connection (options) {
 	this.terminated = false;
 
 	events.EventEmitter.call(this);
+        
+        this.maintainMinConnection();
 }
 
 sysu.inherits(Connection, events.EventEmitter);
 
+
+/**
+ *
+ * @private
+ */
+Connection.prototype.maintainMinConnection = function() {
+    if (typeof (this.options.minConnections - this.sockets.length) === 'number') {
```

Comment:
What is this check guarding against? We know that this.options.minConnections will have a numeric value because it's setup in the initialiser.